### PR TITLE
 fix: use filtered subnet cidr blocks instead of the VPC cidr

### PR DESCRIPTION
**Pull request by bmbferreira**
## what

This PR changes the module to start using the filtered subnet cidrs to create the routes, instead of using the whole VPC cidr.

## why

Currently even when filtering the accepter or requester subnets by tags, the route table on the requester/accepter side is always created to the vpc cidr and doesn't create individual routes for each filtered subnet. This is not flexible enough as we might need for example peering to two different VPCs with overlapping CIDRs and we might want to use the subnet cidrs to be able to route to individual cidrs within the subnet, dodging the VPC cidr overlapping.

### CODE REVIEW

**Comment by bmbferreira** [comment]
@hans-d any plans to merge this? the test failure doesn't seem to be related with my change, as I see the same test failin in other PRs as well: https://github.com/cloudposse/actions/actions/runs/8132588929/job/22223186662

**Comment by hans-d** [comment]
Unfortunately, I cannot bypass this check. Getting some internal visability on these failing tests as well, as they are now popping up in various places (for various different reasons).
Once the blocking bits are gone (soon I hope), this will be revisited.

**Comment by bmbferreira** [comment]
@joe-niland @dotCipher Hi! What can I do to have this reviewed? Thank you

**Comment by bmbferreira** [comment]
@goruha just fixed format that was failing. Can you check if the PR is ok to merge? thank you

**Review by goruha — CHANGES_REQUESTED** [review]
@bmbferreira why do not we have ?

```
data "aws_subnet" "requester" {
```

**Comment by bmbferreira** [comment]
> @bmbferreira why do not we have ?
> 
> ```
> data "aws_subnet" "requester" {
> ```

Fixed. I applied the same logic to the requester as well. I also did an improvement to fallback to the VPC cidr if no tag filters are passed as inputs. I think it only makes sense to use the filtered subnet cidr blocks if we are actually filtering it. If not we can continue using the vpc cidr. WDYT?

File: accepter.tf [hunk 1]
```diff
@@ -58,11 +58,19 @@ data "aws_subnets" "accepter" {
...
+  provider = aws.accepter
+  id       = each.value
+}
+
 locals {
-  accepter_subnet_ids = local.accepter_enabled ? data.aws_subnets.accepter[0].ids : []
-  accepter_vpc_id     = join("", data.aws_vpc.accepter[*].id)
-  accepter_account_id = join("", data.aws_caller_identity.accepter[*].account_id)
-  accepter_region     = join("", data.aws_region.accepter[*].name)
+  accepter_subnet_ids       = local.accepter_enabled ? data.aws_subnets.accepter[0].ids : []
```
Comment (goruha) [inline accepter.tf:68]:
```suggestion
  accepter_subnet_ids       = local.accepter_enabled ? try(data.aws_subnets.accepter[0].ids, []) : []
```

**Review by goruha — CHANGES_REQUESTED** [review]

**Comment by goruha** [comment]
@bmbferreira I have one request for change.
Beside of that it looks fine

**Review by goruha — APPROVED** [review]

**Comment by goruha** [comment]
@bmbferreira Thanks for your contribution

**Pull request merged**